lldb/source/Plugins/ObjectFile/Minidump/MinidumpFileBuilder.cpp

```diff
@@ -797,20 +822,75 @@ void MinidumpFileBuilder::AddLinuxFileStreams(
   }
 }
 
-Status MinidumpFileBuilder::Dump(lldb::FileUP &core_file) const {
-  constexpr size_t header_size = sizeof(llvm::minidump::Header);
-  constexpr size_t directory_size = sizeof(llvm::minidump::Directory);
+Status MinidumpFileBuilder::AddMemory(const ProcessSP &process_sp,
+                                      SaveCoreStyle core_style) {
+  Status error;
+
+  Process::CoreFileMemoryRanges ranges_for_memory_list;
```

Comment:
It might be useful to create two Process::CoreFileMemoryRanges lists: one for 32 bit and one for 64 bit and have a variable that keeps track of what was already added:
```
uint64_t ranges_32_size = 0;
Process::CoreFileMemoryRanges ranges_32;
Process::CoreFileMemoryRanges ranges_64;
```
Then we can add regions to each list as needed below without having to remove anything from the `ranges` list below?

Reply:
I responded above on this, I tried to minimize allocating vectors and reduce iterations by just removing what will fit in 32b.